<launch>
  <arg name="vehicle" default="rasl1"/>
  <arg name="rate" default="1"/>
  <arg name="index" default="0"/>
  <arg name="bagfile" default="simulation"/>
  <rosparam param="robot_list">[rasl1,rasl2,rasl3]</rosparam>
  <rosparam param="robot_id_list">[a,b,c]</rosparam>

  <node pkg="rosbag"
    name="rosbag_simulation_$(arg vehicle)"
    type="play"
    args="-r $(arg rate) -s 5 /home/rasl/bagfiles/velodyne/$(arg vehicle)_velodyne.bag">
  </node>

  <!-- Record important packages for replay or matlab analysis -->
  <node pkg="rosbag"
    type="record"
    name="record"
    output="screen"
    args="-O /home/rasl/bagfiles/two_robots_simulation/$(arg bagfile).bag
    /distributed_mapping_$(arg vehicle)/knn_hist
    /distributed_mapping_$(arg vehicle)/packet_size
    /distributed_mapping_$(arg vehicle)/relpose_estimate
    /distributed_mapping_$(arg vehicle)/scan_points_count
    /distributed_mapping_$(arg vehicle)/match_scan_info
    /distributed_mapping_$(arg vehicle)/timer
    /cpu_monitor/cpu_status
    "/>

  <node name="rviz"
        pkg="rviz"
        type="rviz"
        output="screen"
        args="-d $(find joint_mapping)/rviz/velodyne_visualization.rviz"/>

  <node name="scan_odom_sync"
    pkg="scan_odom_sync"
    type="scan_odom_sync"
    output="screen">

    <param name="robot_name" value="$(arg vehicle)"/>
    <param name="in_cloud_topic" value="/velodyne_points"/>
    <param name="update_interval" value="5" />

    <param name="robot_frame" value="velodyne" />
    <param name="world_frame" value="sensor_init" />

    <remap from="~/$(arg vehicle)/point_cloud_with_pose" to="/$(arg vehicle)/point_cloud_with_pose"/>
  </node>

<!--========================= distributed_mapping nodes ========================-->
  <node pkg="joint_mapping"
    name="distributed_mapping_$(arg vehicle)"
    type="joint_mapping_node"
    output="screen">
    <!-- launch-prefix="gdb -ex run - -args" > -->
    <!-- setting: default distributed_mapping param and agent name -->
    <rosparam file="$(find joint_mapping)/config/parameters.yaml"/>
    <rosparam file="$(find joint_mapping)/config/network_parameters.yaml"/>

    <!-- Index into the robot list -->
    <param name="index" value="$(arg index)"/>
    <param name="use_udp" value="true"/>

    <param name="range_max" value="100"/>
    <param name="robot_frame" value="velodyne"/>
    <param name="world_frame" value="sensor_init" />

    <param name="frame_id/fixed" value="sensor_init"/>
    <param name="frame_id/dma" value="dma" />
    <param name="frame_id/map" value="sensor_init" />
  </node>

</launch>

